src/types/Token.ts

```diff
@@ -1,10 +1,19 @@
-interface Token {
+type CustomTokenPart<T> = {
+  [P in keyof T]: T[P]
+};
+
+type StaticTokenPart = {
   access_token: string;
   expires_in?: number;
   scope?: string[];
   token_type?: string;
   local_expiry?: number;
-  [key: string]: any;
 }
 
-export { Token };
+type Token<T = any> = CustomTokenPart<T> & StaticTokenPart;
```

Comment:
What is the difference between using `CustomTokenPart<T>` vs:

```typescript
type Token<T = any> = T & StaticTokenPart;
```
?

Other than that it looks like it is serving the use case well!